pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -955,6 +956,17 @@ func (c *Controller) AdditionalPodSubscriptions(
 	return shouldSubscribe
 }
 
+// syncAllWorkloads refreshes all ambient workloads.
+func (c *Controller) syncAllWorkloads() {
+	if c.ambientIndex != nil {
+		namespaces := c.namespaces.List(metav1.NamespaceAll, klabels.Everything())
```

Comment:
We should use discoverNamespaceFilter here

Reply:
Great point, updated.